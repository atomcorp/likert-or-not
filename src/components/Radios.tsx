import * as React from 'react';
import {
  isChecked,
  returnValueIfSpecifiedElseEnum,
  returnEnumArray,
} from './utils';
import Radio from './Radio';
import LikertOrNot from '../index.d';

const Radios = (props: LikertOrNot.RadiosProps) => (
  <>
    {returnEnumArray(props.length).map((index) => (
      <td key={index}>
        <Radio
          name={props.id}
          value={returnValueIfSpecifiedElseEnum({
            index,
            values: props.values,
          })}
          checked={isChecked({
            value: props.value,
            values: props.values,
            index,
          })}
          handleClick={props.handleClick}
          header={props.headers[index]}
        />
      </td>
    ))}
  </>
);

export default Radios;
