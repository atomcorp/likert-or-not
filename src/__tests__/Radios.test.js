import React from 'react';
import Radios from '../components/Radios';
import renderer from 'react-test-renderer';

test('Radios outputs an array of radio inputs', () => {
  const component = renderer.create(
    <Radios
      length={3}
      id={'RowId'}
      values={['A', 'B', 'C']}
      value={'A'}
      headers={['Header 1', 'Header 2', 'Header 3']}
      handleClick={jest.fn(() => ({}))}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
