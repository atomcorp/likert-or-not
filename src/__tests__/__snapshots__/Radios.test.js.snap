// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radios outputs an array of radio inputs 1`] = `
Array [
  <td>
    <label
      className={undefined}
    >
      <input
        checked={true}
        className={undefined}
        name="RowId"
        onChange={[Function]}
        type="radio"
        value="A"
      />
      <span
        className={undefined}
      >
        Header 1
      </span>
    </label>
  </td>,
  <td>
    <label
      className={undefined}
    >
      <input
        checked={false}
        className={undefined}
        name="RowId"
        onChange={[Function]}
        type="radio"
        value="B"
      />
      <span
        className={undefined}
      >
        Header 2
      </span>
    </label>
  </td>,
  <td>
    <label
      className={undefined}
    >
      <input
        checked={false}
        className={undefined}
        name="RowId"
        onChange={[Function]}
        type="radio"
        value="C"
      />
      <span
        className={undefined}
      >
        Header 3
      </span>
    </label>
  </td>,
]
`;
